tests/setupconfig/test_steps.py

```diff
@@ -0,0 +1,247 @@
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/certs"
+    )
+    assert (
+        config.oidc_op_authorization_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/auth"
+    )
+    assert (
+        config.oidc_op_token_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/token"
+    )
+    assert (
+        config.oidc_op_user_endpoint
+        == f"{KEYCLOAK_BASE_URL}protocol/openid-connect/userinfo"
+    )
+    assert config.username_claim == ["claim_name"]
+    assert config.groups_claim == ["groups_claim_name"]
+    assert config.claim_mapping == {"first_name": ["given_name"]}
+    assert not config.sync_groups
+    assert config.sync_groups_glob_pattern == "local.groups.*"
+    assert set(group.name for group in config.default_groups.all()) == {
+        "local.groups.Admins",
+        "local.groups.Read-only",
+    }
+    assert config.make_users_staff
+    assert config.superuser_group_names == ["superuser"]
+    assert not config.oidc_use_nonce
+    assert config.oidc_nonce_size == 48
+    assert config.oidc_state_size == 48
+    assert config.userinfo_claims_source == UserInformationClaimsSources.id_token
+
+    assert step.is_configured(setup_config_full_model)
+
+
+@pytest.mark.django_db
+def test_required_settings():
+    output = StringIO()
+    err = StringIO()
+    with pytest.raises(CommandError) as command_error:
+        call_command(
+            "setup_configuration",
+            yaml_file=["tests/setupconfig/files/empty.yml"],
+            stdout=output,
+            stderr=err,
+        )
+
+    assert "Prerequisites for configuration are not fulfilled:" in str(
+        command_error.value
+    )
+
+    assert "ADMIN_OIDC.oidc_rp_client_id" in str(command_error.value)
+    assert "ADMIN_OIDC.oidc_rp_client_secret" in str(command_error.value)
+    assert "ADMIN_OIDC.endpoint_config" in str(command_error.value)
+
+    config = OpenIDConnectConfig.get_solo()
+    assert not config.enabled
+
+
+@pytest.mark.django_db
+def test_partial_endpoints_provided():
+    """
+    Test what if only one endpoint (not discovery) is provided
+    """
+    output = StringIO()
+    err = StringIO()
+    with pytest.raises(CommandError) as command_error:
+        call_command(
+            "setup_configuration",
+            yaml_file=["tests/setupconfig/files/partial_endpoints.yml"],
+            stdout=output,
+            stderr=err,
+        )
+
+    assert "Prerequisites for configuration are not fulfilled:" in str(
+        command_error.value
+    )
+
+    assert (
+        "ADMIN_OIDC.endpoint_config.OIDCFullEndpointConfig.oidc_op_token_endpoint"
+        in str(command_error.value)
+    )
+    assert (
+        "ADMIN_OIDC.endpoint_config.OIDCFullEndpointConfig.oidc_op_user_endpoint"
+        in str(command_error.value)
+    )
+
+    assert (
+        "ADMIN_OIDC.endpoint_config.OIDCDiscoveryEndpoint.oidc_op_discovery_endpoint"
+        in str(command_error.value)
+    )
```

Comment:
It reports all missing fields instead of the ones from the model `OIDCFullEndpointConfig` when one field from that is provided.

Reply:
Fxied